client/cache/redis/redis.go

```diff
@@ -204,60 +209,93 @@ func (rc *Cache) Scan(pattern string) (keys []string, err error) {
 // config: must be in this format {"key":"collection key","conn":"connection info","dbNum":"0", "skipEmptyPrefix":"true"}
 // Cached items in redis are stored forever, no garbage collection happens
 func (rc *Cache) StartAndGC(config string) error {
-	var cf map[string]string
+	err := rc.parseConf(config)
+	if err != nil {
+		return err
+	}
+
+	rc.connectInit()
+
+	c := rc.p.Get()
+	defer func() {
+		_ = c.Close()
+	}()
+
+	// test connection
+	if err = c.Err(); err != nil {
+		return berror.Wrapf(err, cache.InvalidConnection,
+			"can not connect to remote redis server, please check the connection info and network state: %s", config)
+	}
+	return nil
+}
+
+func (rc *Cache) parseConf(config string) error {
+	var cf redisConfig
 	err := json.Unmarshal([]byte(config), &cf)
 	if err != nil {
 		return berror.Wrapf(err, cache.InvalidRedisCacheCfg, "could not unmarshal the config: %s", config)
 	}
 
-	if _, ok := cf["key"]; !ok {
-		cf["key"] = DefaultKey
-	}
-	if _, ok := cf["conn"]; !ok {
-		return berror.Wrapf(err, cache.InvalidRedisCacheCfg, "config missing conn field: %s", config)
+	err = cf.parse()
+	if err != nil {
+		return err
 	}
 
+	rc.dbNum = cf.DbNum
+	rc.key = cf.Key
+	rc.conninfo = cf.Conn
+	rc.password = cf.password
+	rc.maxIdle = cf.MaxIdle
+	rc.idleTimeout = cf.idleTimeout
+	rc.skipEmptyPrefix = cf.SkipEmptyPrefix
+
+	return nil
+}
+
+type redisConfig struct {
+	DbNum           int    `json:"dbNum"`
+	SkipEmptyPrefix bool   `json:"skipEmptyPrefix"`
+	Key             string `json:"key"`
 	// Format redis://<password>@<host>:<port>
-	cf["conn"] = strings.Replace(cf["conn"], "redis://", "", 1)
-	if i := strings.Index(cf["conn"], "@"); i > -1 {
-		cf["password"] = cf["conn"][0:i]
-		cf["conn"] = cf["conn"][i+1:]
-	}
+	Conn           string `json:"conn"`
+	MaxIdle        int    `json:"maxIdle"`
+	IdleTimeoutStr string `json:"timeout"`
+
+	// parse from Conn
+	password string
+	// parse from IdleTimeoutStr
+	idleTimeout time.Duration
```

Comment:
while use `Timeout time.Duration` you don't need this field, and please remove it